mars_troughs/accumulation_model.py

```diff
@@ -31,41 +28,203 @@ class InsolationAccumulationModel(AccumulationModel):
     """
 
-    def __init__(self, times: np.ndarray, insolations: np.ndarray):
-        self._ins_times = times
-        self._insolations = insolations
-        self._ins_spline = IUS(self._ins_times, self._insolations)
-        self._ins_spline_integ = self._ins_spline.antiderivative()
-        self._ins2_spline = IUS(self._ins_times, self._insolations ** 2)
-        self._ins2_spline_integ = self.ins2_spline.antiderivative()
+    def __init__(self, times: np.ndarray, insolation: np.ndarray):
+        
+        self.ins_data_spline = IUS(times, insolation)
+        self.iins_data_spline = self.ins_data_spline.antiderivative()
+        self.ins2_data_spline = IUS(times, insolation ** 2)
+        self.iins2_data_spline = self.ins2_data_spline.antiderivative()
+        
+    def get_xt(self, time: np.ndarray, iretreat_model_t_spline: np.ndarray, 
+                                                       angle: float = 2.9):  
+        """
+        Calculates the horizontal distance x (in m) traveled by a point in the
+        center of the high side of the trough. This distance x is a function of 
+        the accumulation rate A and the retreat rate of ice R
+        as in dx/dt=(R(l(t),t)+A(ins(t))cos(theta))/sin(theta). Where theta
+        is the angle of the trough.
+
+        Args:
+            time (np.ndarray): times at which we want the path.
+        Output:
+            horizontal distances (np.ndarray) of the same size as time input.
+        """
+        self.angle=angle           
+        yt = self.get_yt(time)
+        
+        return -self.cot_angle * yt + self.csc_angle * (
+         iretreat_model_t_spline(time) - iretreat_model_t_spline(0))
+    
+    @property
+    def angle(self) -> float:
+        """
+        Slope angle in degrees.
+        """
+        return self._angle * 180.0 / np.pi
+
+    @angle.setter
+    def angle(self, value: float) -> float:
+        """Setter for the angle"""
+        self._angle = value * np.pi / 180.0
+        self._csc = 1.0 / np.sin(self._angle)
+        self._cot = np.cos(self._angle) * self._csc
 
+    @property
+    def csc_angle(self) -> float:
+        """
+        Cosecant of the slope angle.
+        """
+        return self._csc
 
-class LinearInsolationAccumulation(InsolationAccumulationModel):
+    @property
+    def cot_angle(self) -> float:
+        """
+        Cotangent of the slope angle.
+        """
+        return self._cot
+
+class LinearAccuIns(AccumulationDepInsolation):
     """
     Accumulation is linear in solar insolation.
 
     Args:
         times (np.ndarray): times at which the solar insolation is known
         insolations (np.ndarray): value of the solar insolations
-        intercept (float, optional): default is 0 millimeters
-        slope (float, optional): default is 1e-6 mm per unit
-            of solar insolation square.
+        intercept (float, optional): default is 0 m
+        slope1 (float, optional): default is 1e-6 m per unit
```

Comment:
If this is called `slope1` in the docstring, why is it called `slope` in the signature?

Reply:
it should be called slope everywhere for the linear accumulation model